sycl/source/detail/scheduler/commands.cpp

```diff
@@ -876,15 +884,24 @@ cl_int ExecCGCommand::enqueueImp() {
 
     // Remember this information before the range dimensions are reversed
     const bool HasLocalSize = (NDRDesc.LocalSize[0] != 0);
+    const auto OrigNDRDesc = NDRDesc;
 
     ReverseRangeDimensionsForKernel(NDRDesc);
 
-    PI_CALL(RT::piEnqueueKernelLaunch, MQueue->getHandleRef(), Kernel,
-            NDRDesc.Dims, &NDRDesc.GlobalOffset[0], &NDRDesc.GlobalSize[0],
-            HasLocalSize ? &NDRDesc.LocalSize[0] : nullptr, RawEvents.size(),
-            RawEvents.empty() ? nullptr : &RawEvents[0], &Event);
-
-    return PI_SUCCESS;
+    pi_result Error = PI_CALL_RESULT(
+        RT::piEnqueueKernelLaunch, MQueue->getHandleRef(), Kernel, NDRDesc.Dims,
+        &NDRDesc.GlobalOffset[0], &NDRDesc.GlobalSize[0],
+        HasLocalSize ? &NDRDesc.LocalSize[0] : nullptr, RawEvents.size(),
+        RawEvents.empty() ? nullptr : &RawEvents[0], &Event);
+
+    // Analyze return code in order to convert pi_error to a corresponding
+    // exception describing the possible problem
+    pi_device Device =
+        detail::getSyclObjImpl(MQueue->get_device())->getHandleRef();
+    if (detail::enqueue_kernel_launch::handleError(Error, Device, Kernel,
+                                                   OrigNDRDesc))
```

Comment:
Why are you using not "reversed" NDRDesc here?

Reply:
I don't remember the exact reason, but this was incorrect, fixed